src/runtime/exceptions.cs

```diff
@@ -89,7 +89,8 @@ public new static IntPtr tp_str(IntPtr ob)
         static void ClearOffsetHelper(IntPtr ob, int offset)
         {
             var field = Marshal.ReadIntPtr(ob, offset);
-            Runtime.Py_CLEAR(ref field);
+            Runtime.XDecref(field);
+            Marshal.WriteIntPtr(ob, offset, IntPtr.Zero);
```

Comment:
Actually, you must null the field before calling `DecRef`, otherwise decref might eventually call code, that will access the field and read value, that is already dead.

This is the main purpose of `Py_CLEAR` macro in the CPython codebase.